tfe/provider.go

```diff
@@ -574,6 +610,24 @@ func checkConstraints(c *disco.Constraints) error {
 	return fmt.Errorf("%s\n\n%s", summary, details)
 }
 
+func validateTFEVersionStr(v string) error {
+	// If the remote version fails this validation something went terribly wrong
+	// on TFE's end.
+	if v == "" {
+		return fmt.Errorf("can not be an empty string")
+	}
+
+	if v[0] != 'v' {
+		return fmt.Errorf("version provided must be prefixed with v")
+	}
+
+	if len(strings.Split(v[1:], "-")) != 2 {
+		return errors.New("version must hyphenate release number")
+	}
+
+	return nil
```

Comment:
This looks like it could be simplified, both in terms of the procedure and the diversity of errors.

https://go.dev/play/p/j9YMiftacPc